### PR TITLE
Expose current frame in AnimationPlayer2D

### PR DESCRIPTION
**Pull request by PraxTube**
Also renamed the main example because I didn't even realize that was supposed to be the main example (which also confused me the very first time I tried trickfilm out). Not sure if demo is the correct name, but I would kinda highlight that that is the example you should look at first.

I would also like to test more of the features, for example the `just_finished` stuff. The problem is that animations are by nature hard to test in headless, so examples are the only we can do "integration" tests. Because otherwise I always wind up testing the new features in my projects which isn't ideal haha.

For example I am not sure if the implementation of the frames even covers all the edge cases (though it should?).

### CODE REVIEW

**Comment by PraxTube** [comment]
Hm I actually don't know if this simple solution is so great. I also don't know if it might not be nice to keep track of whether the current frame is new (i.e. if we just changed frames). Because we are gonna see the same frame a couple times before the next one.

**Comment by KirmesBude** [comment]
Will take a closer look sometime soon. Will also prototype the callbacks which might be what you are looking for in regards to new frame detection.

File: examples/demo.rs [hunk 1]
```diff
@@ -78,3 +93,17 @@ fn keyboard_animation_control(
...
+    q_animation_player: Query<&AnimationPlayer2D>,
+) {
+    let Ok(mut text) = q_frame_text.get_single_mut() else {
+        return;
+    };
+    let Ok(animation_player) = q_animation_player.get_single() else {
+        return;
+    };
+
+    text.sections[0].value = format!(" current frame: {}", animation_player.frame());
```
Comment (KirmesBude) [inline examples/demo.rs]: Leading space.
Also inconsistent with the initial text from line 36 - I think the Should be "current frame: 0"

File: src/animation/mod.rs [hunk 2]
```diff
@@ -37,6 +37,7 @@ struct PlayingAnimation2D {
     repeat: RepeatAnimation,
     speed: f32,
     elapsed: f32,
+    frame: usize,
```
Comment (KirmesBude) [inline src/animation/mod.rs:40]: Fells wrong to have this on here considering we have all the information already here. But unless we ask the user to supply `Res<Assets<AnimationClip2D>>` this is the best we can do.

Comment (PraxTube) [reply src/animation/mod.rs:40]: Yeah I know what you mean, if you have ideas for a better implementation feel very free to edit this PR. I needed this in my game which is why I just took the easiest way.

**Comment by KirmesBude** [comment]
Sorry for the delay. I am inclined to merge this as is for now.
I want to experiment with the Animatable trait on bevy main and see if we get this for free there.

**Comment by PraxTube** [comment]
No worries, yeah fine by me. I wasn't aware of the Animatable trait, that might be interesting to look into.
